### PR TITLE
Add audit log recording and retrieval

### LINKED ISSUE #128: Audit Log
Similar feature to what we have in Netlify. We need to track events in a new table and offer a UI for Netlify Site owners to see activity log. This is related to https://github.com/netlify/gotrue/pull/124, because that table should also use JSON to better indexing arbitrary data.

### PR DESCRIPTION
**Pull request by brycekahle**
**- Summary**

Add audit log entries for important actions to be viewable by admins. Fixes #128 

**- Test plan**

Added simple tests to exercise new endpoint.

**- Description for the changelog**

Add audit log recording and retrieval.

### CODE REVIEW

File: api/audit.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/netlify/gotrue/models"
+)
+
+var filterColumnMap = map[string][]string{
+	"author": []string{"actor_email", "actor_name"},
```
Comment (bcomnes) [inline api/audit.go]: Just to confirm I'm reading this right, there is only one query right now (author) ?

File: models/audit_log_entry.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	UserModifiedAction:          user,
+	UserRecoveryRequestedAction: user,
+}
+
+// AuditLogEntry is the database model for audit log entries.
+type AuditLogEntry struct {
+	InstanceID uuid.UUID `json:"-" db:"instance_id"`
+	ID         uuid.UUID `json:"id" db:"id"`
+
+	Payload JSONMap `json:"payload" db:"payload"`
```
[the remark this answers is not in the document]
Comment (bcomnes) [reply models/audit_log_entry.go]: Feel free to do it however works best.  The payload pattern is just what the team audit log does.

File: api/audit.go (hunk 1 above)
Comment (brycekahle) [reply api/audit.go]: Yes. It didn't sound like we had any others that would apply at the moment.

File: models/audit_log_entry.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (brycekahle) [reply models/audit_log_entry.go]: I wanted to match the payload format @bcomnes provided to eliminate the need for a translation layer.

File: api/audit.go (hunk 1 above)
Comment (bcomnes) [reply api/audit.go]: 👍

**Pull request merged**